### ISSUE TITLE
Texture display wrong in IOS browsers

### ISSUE THREAD

**Issue by ehipasika** [stephomi/sculptgl#47]
I found in ScupltGL can not display correctly in all IOS browers, but IOS claims it support WEBGL, and i test another webgl matcap sample base on matcap, and it's work fine.
pls check this link for test video:
https://vimeo.com/185645917

**Comment by stephomi** [comment]
Thanks for the report, unfortunately I have an android,, so for now I can't really debug it.

**Comment by as02700** [comment]
something wrong with me too.
when i // if (this._showGrid) this._grid.render(this)
the texture can not display correctly( look like color lump / color spot ).
the video : https://youtu.be/YX-jsQu-4gI

**Comment by tycho01** [comment]
[This](https://github.com/stephomi/sculptgl/commit/2fe892267c3ac66bc006c253c4857b8fc9857ed6) commit appeared to have broken the shader on some iOS environments.

**Comment by stephomi** [comment]
Thanks, I suspect something wrong with [shader derivatives](https://github.com/stephomi/sculptgl/blob/master/src/render/shaders/glsl/curvature.glsl#L6), can you try https://goo.gl/m9C6Bx and see if it works?

**Comment by as02700** [comment]
yes, i work ( https://goo.gl/m9C6Bx ). seen like as desktop

**Comment by ehipasika** [comment]
Yes, it works fine in IOS10 Chrome. So if it means your suspect not established? 
In my further test for SculptGL shader in IOS10 Chrome, I found PBR,matcap,normal shader are also not work well, but UV shader is OK. 
So I guess the bug is from Reflection/Environment Mapping, pls check Jaume Sanchez's blog about Spherical reflection (or environment) mapping (SEM) 
 [https://www.clicktorelease.com/blog/creating-spherical-environment-mapping-shader](url
Someone also report a bug in IOS8, and the solve the problem, but Jaume Sanchez's tutorial is base on three.js framework, I don't know if it work in native webgl.

**Comment by stephomi** [comment]
I'm almost sure it's not the matcap shader as PBR and normal have the issue too.
It can be many things, but until I debug on an iOS for real, it's tricky to solve :p.

**Comment by as02700** [comment]
i came to ( https://www.browserstack.com/ ) test on IOS online within a free account.
it can test on locale need a plugin

**Comment by stephomi** [comment]
Ok I'll see if I can reproduce and if it's not too complex to setup!

**Comment by selfsame** [comment]
I believe this was fixed with 8414c652afb00beb680b89f64aa33b019399b34c
